core/src/main/java/com/datastax/oss/driver/internal/core/CqlIdentifiers.java

```diff
@@ -16,21 +16,36 @@
 package com.datastax.oss.driver.internal.core;
 
 import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.shaded.guava.common.base.Preconditions;
 import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableList;
 import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableMap;
+import edu.umd.cs.findbugs.annotations.NonNull;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Function;
 
 public class CqlIdentifiers {
 
-  public static List<CqlIdentifier> wrap(Iterable<String> in) {
+  public static List<CqlIdentifier> wrap(
+      @NonNull Iterable<String> in, @NonNull Function<String, CqlIdentifier> fn) {
```

Comment:
You could keep this private for now if it's not used anywhere.

Reply:
I considered that, but there wasn't a glaring reason to restrict it either... so I just kept it public to maximize API options.

Reply:
If there's no reason one way or another, I prefer being conservative. It's easier to expand the visibility later than restrict it.

Reply:
I tend to take the opposite approach: exposing more API surface area provides greater flexibility for other developers.  But I'll defer here in the interest of getting this PR wrapped up.